### ISSUE TITLE
Undefined method on function prototype

### ISSUE THREAD

**Issue by 0x0049** [NeilFraser/JS-Interpreter#137]
I'm not sure whether this is actually a bug or some sort of fundamental misunderstanding on my part about prototypes or functions or maybe the capabilities of the interpreter. My use case is inheritance with a parent/child class type of deal.

```
function Example() {}
Example.prototype.test = function () {
	return "Example";
};

const code = `
	Example.prototype.test();
`;

console.log(eval(code));

const interpreter = new Interpreter(code, (intp, scope) => {
	intp.setProperty(scope, "Example", intp.nativeToPseudo(Example));
});

interpreter.run();

console.log(interpreter.value);
```

The eval & interpreter pair above output:

```
Example
undefined is not a function
```

I thought I might just try putting `Example` inline, which lead me to a difference when using `for...in`.

```
const code = `
	function Example() {}
	console.log("for p in Example:")
	for (var p in Example) {
		console.log(p);
	}
`;

eval(code);

const interpreter = new Interpreter(code, (intp, scope) => {
	intp.setProperty(scope, "console", intp.nativeToPseudo({
		log: (...args) => console.log(...args),
	}));
});
interpreter.run();
```

This results in:

```
for p in Example:
for p in Example:
prototype
length
```

The `for...in` is used in my use case to extend `Example`. I believe it's part of the code Typescript generates (actually it might be Babel) when it extends classes. In any case, in strict mode this ends up resulting in an error when trying to copy `length`.

I also wanted to thank you for creating the interpreter! It's awesome and I'm stoked to make something cool with it.

**Comment by cpcallen** [comment]
So, in the first example (abridged):

```JS
function Example() {}
Example.prototype.test = function () {
	return "Example";
};

const interpreter = new Interpreter(code, (intp, scope) => {
	intp.setProperty(scope, "Example", intp.nativeToPseudo(Example));
});
```

In this case you will successfully create a function named `Example`, but because of the way `nativeToPseudo` is written it won't do what you want on two counts:

1. `nativeToPseudo` only copies properties of plain `Object` and `Array` objects.  `Example` is a function, so the "pseudo" version of it won't have a `.prototype` property, and so of course there will be no `.prototype.test` method to call.  This is arguably a bug, and wouldn't be too difficult to fix.  However…

1. The pseudo-function objects returned by `nativeToPseudo` are always marked as non-constructible, so even the point above was fixed, allowing you to call `Example.prototype.test()`, it would still not be possible to do `(new Example).test()`.

The ability to apply `nativeToPseudo` to functions was really a convenience to simplify certain trivial uses of `createNativeFunction`.  Both are really intended for installing APIs to access the surrounding native APIs (like `console.log`), not as conversion functions for plain JS code.

Your second example shows how we would expect users to create ordinary JS functions within the interpreter (again abridged):

```JS
const code = `
	function Example() {}
	console.log("for p in Example:")
	for (var p in Example) {
		console.log(p);
	}
`;
const interpreter = new Interpreter(code, /* ... */);
interpreter.run();
```

In this case the discrepant behaviour is caused by the `.length` and `.prototype` properties being created as innumerable.  I have filed #138 about this; it should be very easy to fix.
